Dispatcher/UI/Homepage/HomepageViewController.swift

```diff
@@ -23,12 +27,35 @@ class HomeViewController: UIViewController {
         articlesTableView.dataSource = self
         articlesTableView.delegate = self
         
-        articlesTableView.register(UINib(nibName: "ArticleCell", bundle: nil), forCellReuseIdentifier: ArticleCell.articleCellIdentifier)
-        
+        articlesTableView.register(UINib(nibName: AppConstants.articleCellNibName, bundle: nil), forCellReuseIdentifier: AppConstants.articleCellIdentifier)
+        configureActivityIndicator()
+        fetchArticles()
     }
     
+    private func configureActivityIndicator() {
+        activityIndicator.style = .large
+        activityIndicator.center = homepageView.center
+        homepageView.addSubview(activityIndicator)
+    }
+    
+    private func fetchArticles() {
+        self.activityIndicator.startAnimating()
+        homepageViewModel.getTopArticlesFromAPI(pageNumber: currentPage, completionHandler: { errorMsg in
+            if(errorMsg != nil) {
+                self.present(Utils.showErrorAlert(errorMsg!),animated: true, completion: nil)
+                self.activityIndicator.stopAnimating()
+            } else {
+                DispatchQueue.main.async {
+                    self.activityIndicator.stopAnimating()
+                    self.currentPage += 1
```

Comment:
Should happen inside the viewModel once we know we fetched successfully and added the data to our array